astroquery/esa_hubble/core.py

```diff
@@ -0,0 +1,396 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+
+@author: Javier Duran
+@contact: javier.duran@sciops.esa.int
+
+European Space Astronomy Centre (ESAC)
+European Space Agency (ESA)
+
+Created on 13 Ago. 2018
+
+
+"""
+from astroquery.utils import commons
+from astropy import units
+from astropy.units import Quantity
+from astroquery.utils.tap.core import TapPlus
+from astroquery.utils.tap.model import modelutils
+from astroquery.query import BaseQuery
+
+from . import conf
+from astropy import log
+
+
+__all__ = ['EsaHubble', 'EsaHubbleClass']
+
+
+class ESAHubbleHandler(BaseQuery):
+
+    def __init__(self):
+        super(ESAHubbleHandler, self).__init__()
+
+    def get_file(self, filename, response, verbose=False):
+        file = open(filename, 'wb')
+        file.write(response.content)
+        file.close()
+
+        if "/" not in filename:
+            log.info("File " + str(filename) + " downloaded to current "
+                     "directory")
+        else:
+            log.info("File " + str(filename) + " downloaded")
+
+    def get_table(self, filename, response, output_format='votable',
+                  verbose=False):
+        file = open(filename, 'wb')
+        file.write(response.content)
+        file.close()
+
+        table = modelutils.read_results_table_from_file(filename,
+                                                        str(output_format))
+        return table
+
+    def request(self, t="GET", link=None, params=None):
+        return self._request(method=t, url=link, params=params)
```

Comment:
Why did you add this extra layer of wrapper?  This looks like unnecessary added complexity to me

Reply:
What I aimed with the handlers is to provide a mean to separate the core behavior of the module from the accesses to the real services. It is a way to mock the services for testing. This way I can have a default handler for accessing the real service and a dummy handler (that does nothing) for testing.
May I ask you, for the time being, to accept that? I can refactor the code as you please in the upcoming updates. Doing this now will impact our schedule.
Is that ok with you?